src/http-request/leave.request.ts

```diff
@@ -0,0 +1,47 @@
+import { EmployeeLeave, LeaveEnum } from '../types/types'
+
+export const leaveData: EmployeeLeave[] = [
+  {
+    employeeId: 1,
+    leave: [
+      { creditBalance: 20, leaveType: LeaveEnum.SL },
+      { creditBalance: 20, leaveType: LeaveEnum.SLP },
+      { creditBalance: 20, leaveType: LeaveEnum.VL },
+    ],
+  },
+  {
+    employeeId: 2,
+    leave: [
+      { creditBalance: 20, leaveType: LeaveEnum.SL },
+      { creditBalance: 20, leaveType: LeaveEnum.SLP },
+      { creditBalance: 20, leaveType: LeaveEnum.VL },
+    ],
+  },
+  {
+    employeeId: 3,
+    leave: [
+      { creditBalance: 20, leaveType: LeaveEnum.SL },
+      { creditBalance: 20, leaveType: LeaveEnum.SLP },
+      { creditBalance: 20, leaveType: LeaveEnum.VL },
+    ],
+  },
+]
+
+export const getLeaveCredits = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(leaveData)
+    }, 1000)
+  })
+}
+
+export const getLeaveById: (id: number) => Promise<EmployeeLeave> = (id) => {
```

Comment:
- [ ] Change Function Name to getLeaveCreditsByEmployeeId

Reply:
done na po